packages/sdk/src/actor/actor.ts

```diff
@@ -838,24 +856,44 @@ export class Actor implements ActorLike, Patchable<ActorLike> {
 	}
 
 	private _setCollider(collider: Partial<ColliderLike>) {
-		const oldCollider = this._collider;
-		if (this._collider) {
-			unobserve(this._collider);
-			this._collider = undefined;
+		let size = null;
+		let center = null;
+
+		if (collider.geometry.shape === ColliderType.Box) {
+			size = collider.geometry.size;
+			center = collider.geometry.center;
+		} else if (collider.geometry.shape === ColliderType.Sphere) {
+			size = collider.geometry.radius;
+			center = collider.geometry.center;
+		} else if (collider.geometry.shape === ColliderType.Capsule) {
+			size = collider.geometry.size;
+			center = collider.geometry.center;
 		}
 
-		this._collider = new Collider(this, collider);
-		if (oldCollider) {
-			this._collider.internal.copyHandlers(oldCollider.internal);
-		}
+		const geometry = this.generateColliderGeometry(collider.geometry.shape, size, center)
```

Comment:
What is the purpose of moving the collider geometry unpacking to here rather than in the main setCollider function?

Reply:
@tombuMS  - sorry about the pause in the action.  I accidentally left the post (above) in draft mode, and never sent it.  It was a few months ago, but I recall trying your suggestion, but I couldn't get it pass through the TS checks.

Reply:
@tombuMS  - any additional feedback or suggestions on this?